### PR TITLE
NyxLex - Login modal

### PR DESCRIPTION
**Pull request by NyxLex**

### CODE REVIEW

**Review by Percifalll — APPROVED** [review]

File: src/main/java/com/ss/ita/rozetka/pageobject/modals/LoginModal.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import com.ss.ita.rozetka.pageobject.elements.Header;
+import io.qameta.allure.Step;
+
+import static com.codeborne.selenide.Selenide.$;
+import static com.codeborne.selenide.Selenide.$x;
+
+public class LoginModal {
+
+    @Step("LoginModal: close login modal")
+    public Header closeLoginModal() {
```
Comment (toleksyn) [inline src/main/java/com/ss/ita/rozetka/pageobject/modals/LoginModal.java:14]: close()

File: src/main/java/com/ss/ita/rozetka/pageobject/modals/LoginModal.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    }
+
+    @Step("LoginModal: get password visibility toogle")
+    private SelenideElement getPasswordVisibilityToogle() {
+        return $x("//button[@class='button_type_link form__toggle-password']");
+    }
+
+    @Step("LoginModal: set login {login}")
+    public LoginModal setLogin(String login) {
+        SelenideElement input = $x("//input[@type='email']");
```
Comment (toleksyn) [inline src/main/java/com/ss/ita/rozetka/pageobject/modals/LoginModal.java:26]: setLoginEmail()
inputField?

File: src/main/java/com/ss/ita/rozetka/pageobject/modals/LoginModal.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        input.clear();
+        input.sendKeys(login);
+        getPasswordVisibilityToogle().click();
+        return this;
+    }
+
+    @Step("LoginModal: set password {password}")
+    public LoginModal setPassword(String password) {
+        SelenideElement input = $("#auth_pass");
```
Comment (toleksyn) [inline src/main/java/com/ss/ita/rozetka/pageobject/modals/LoginModal.java:35]: inputField?

File: src/test/java/com/ss/ita/rozetka/test/LoginModalTest.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        LoginModal loginModal = new HomePage()
+                .open()
+                .getHeader()
+                .openLoginModal()
+                .setLogin("default")
+                .setPassword(StringUtils.EMPTY);
+
+        assertThat(loginModal.isInvalidLoginIconVisible())
+                .as("visibility of invalid login icon should be equal true")
+                .isEqualTo(true);
```
Comment (toleksyn) [inline src/test/java/com/ss/ita/rozetka/test/LoginModalTest.java:22]: isTrue()
same for all

File: src/test/java/com/ss/ita/rozetka/test/LoginModalTest.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+        assertThat(loginModal.isInvalidPasswordIconVisible())
+                .as("visibility of invalid password icon should be equal true")
+                .isEqualTo(true);
+
+        loginModal.setLogin("test@gmail.com").setPassword("1111");
```
Comment (toleksyn) [inline src/test/java/com/ss/ita/rozetka/test/LoginModalTest.java:28]: format

**Review by toleksyn — CHANGES_REQUESTED** [review]

**Review by NazarYakh — APPROVED** [review]

**Pull request merged**